src/sage/algebras/vertex_operators.py

```diff
@@ -0,0 +1,905 @@
+
+        sage: from sage.algebras.vertex_operators import BosonicFockSpace
+        sage: B = BosonicFockSpace(); B
+        Univariate Laurent Polynomial Ring in w over Symmetric Functions
+        over Rational Field in the Schur basis
+        sage: B.an_element()
+        s[]*w
+        sage: Sym = SymmetricFunctions(QQ); Sym.inject_shorthands(verbose=False)
+        sage: BosonicFockSpace(QQ, s)
+        Univariate Laurent Polynomial Ring in w over
+        Symmetric Functions over Rational Field in the Schur basis
+        sage: BosonicFockSpace(FractionField(QQ['q','t']), 'p')
+        Univariate Laurent Polynomial Ring in w over Symmetric Functions
+        over Fraction Field of Multivariate Polynomial Ring in q, t over
+        Rational Field in the powersum basis
+    """
+    Sym = SymmetricFunctions(R)
+    if isinstance(sym_basis, str):
+        b = getattr(Sym, sym_basis, None)
+        if b is None or b() not in Sym.realizations():
+            raise ValueError("Unknown symmetric function basis")
+        return LaurentPolynomialRing(b(), names=(name,))
+    if sym_basis.base_ring() is not R:
+        raise ValueError("")
+    return LaurentPolynomialRing(sym_basis, names=(name,))
+
+
+class HalfVertexOperator(SageObject):
+    r"""
+    A half vertex operator.
+
+    INPUT:
+
+    - ``f`` -- A function from the positive integers to an infinite weyl algebra
+
+    OUTPUT:
+
+    A stream where the `i`'th value is the coefficient of `z^i` in `\exp \left(\sum_{j \geq 1} f(j) z^j\right)`
+
+    EXAMPLES::
+
+        sage: from sage.algebras.vertex_operators import HalfVertexOperator
+        sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo)
+        sage: H = HalfVertexOperator(lambda n: x[n])
+        sage: H[3]
+        1/6*x[1]^3 + x[1]*x[2] + x[3]
+    """
+    def __init__(self, f):
+        r"""
+        Initialize ``self``.
+
+        TESTS::
+
+            sage: from sage.algebras.vertex_operators import HalfVertexOperator
+            sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo)
+            sage: H = HalfVertexOperator(lambda n: x[n])
+            sage: TestSuite(H).run(skip='_test_pickling')
+        """
+        self._stream = Stream_cauchy_compose(
+            Stream_function(lambda n: ZZ(1)/factorial(n), False, 0),
+            Stream_function(f, False, 1),
+            False
+        )
+
+    def __getitem__(self, i):
+        r"""
+        Get the i'th value of ``self``
+
+        sage: from sage.algebras.vertex_operators import HalfVertexOperator
+        sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo); dx = W.differentials()
+        sage: H = HalfVertexOperator(lambda n: x[n] + dx[n])
+        sage: H[2]
+        1/2*dx[1]^2 + x[1]*dx[1] + dx[2] + 1/2*x[1]^2 + x[2] + 1/2
+        sage: H[-1]
+        Traceback (most recent call last):
+        ...
+        ValueError: Invalid input
+        """
+        if i in NonNegativeIntegers():
+            return self._stream[i]
+        raise ValueError("Invalid input")
```

Comment:
```suggestion
        return ZZ.zero()
```
I prefer this better than an error message, but it is up to you (if an error, then lower case first letter an perhaps an `IndexError` instead?).